package com.example.listycity

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Scaffold
import androidx.compose.ui.Modifier
import com.example.listycity.ui.theme.ListyCityTheme

class MainActivity : ComponentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)

        enableEdgeToEdge()

        val cityRepository = CityRepository()

        setContent {

            ListyCityTheme {

                Scaffold(
                    modifier = Modifier.fillMaxSize()
                ) { innerPadding ->

                    CityListScreen(
                        cities = cityRepository.cities,

                        onAddCity = { city ->
                            cityRepository.addCity(city)
                        },

                        onUpdateCity = { oldCity, updatedCity ->
                            cityRepository.updateCity(
                                oldCity,
                                updatedCity
                            )
                        },

                        onDeleteCity = { city ->
                            cityRepository.deleteCity(city)
                        },

                        modifier = Modifier.padding(innerPadding)
                    )
                }
            }
        }
    }
}